eservice/pdo/eservice/scripts/EServiceEnclaveInfoCLI.py

```diff
@@ -0,0 +1,128 @@
+
+import os
+import sys
+import argparse
+
+import pdo.common.config as pconfig
+import pdo.common.logger as plogger
+
+import pdo.eservice.pdo_helper as pdo_enclave_helper
+import pdo.eservice.pdo_enclave as pdo_enclave
+import logging
+logger = logging.getLogger(__name__)
+
+## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+def LocalMain(config) :
+    # enclave configuration is in the 'EnclaveConfig' table
+    try :
+        logger.debug('initialize the enclave')
+        pdo_enclave_helper.initialize_enclave(config.get('EnclaveModule'))
+        
+        logger.info("MRENCLAVE: " + pdo_enclave.get_enclave_measurement())
+        logger.info("BASENAME: " + pdo_enclave.get_enclave_basename())
+        logger.info("EPID Group: " + pdo_enclave.get_enclave_epid_group())
+        pdo_enclave.dump_enclave_ias_settings()
+    except Error as e :
+        logger.exception('failed to initialize enclave; %s', e)
+        sys.exit(-1)
+
+    sys.exit(0)
+
+## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+## XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+## -----------------------------------------------------------------
+ContractHost = os.environ.get("HOSTNAME", "localhost")
+ContractHome = os.environ.get("CONTRACTHOME") or os.path.realpath("/opt/pdo")
+ContractEtc = os.environ.get("CONTRACTETC") or os.path.join(ContractHome, "etc")
+ContractKeys = os.environ.get("CONTRACTKEYS") or os.path.join(ContractHome, "keys")
+ContractLogs = os.environ.get("CONTRACTLOGS") or os.path.join(ContractHome, "logs")
+ContractData = os.environ.get("CONTRACTDATA") or os.path.join(ContractHome, "data")
+LedgerURL = os.environ.get("LEDGER_URL", "http://127.0.0.1:8008/")
+ScriptBase = os.path.splitext(os.path.basename(sys.argv[0]))[0]
+
+config_map = {
+    'base' : ScriptBase,
+    'data' : ContractData,
+    'etc'  : ContractEtc,
+    'home' : ContractHome,
+    'host' : ContractHost,
+    'keys' : ContractKeys,
+    'logs' : ContractLogs,
+    'ledger' : LedgerURL
+}
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def Main() :
+    # parse out the configuration file first
+    conffiles = [ 'eservice.toml' ]
+    confpaths = [ ".", "./etc", ContractEtc ]
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--config', help='configuration file', nargs = '+')
+    parser.add_argument('--config-dir', help='configuration file', nargs = '+')
+
+    parser.add_argument('--identity', help='Identity to use for the process', required = True, type = str)
+
+    parser.add_argument('--logfile', help='Name of the log file, __screen__ for standard output', type=str)
+    parser.add_argument('--loglevel', help='Logging level', type=str)
+
+    options = parser.parse_args()
+
```

Comment:
Presumably there is sufficient support for parameters to run the script before the configuration is complete?